File: TodoApi/Models/VesselVisitNotifications/Domain/VesselVisitNotification.cs
using System;
using System.Collections.Generic;

namespace TodoApi.Models.VesselVisitNotifications
{
    public class VesselVisitNotification
    {
        public long Id { get; set; }
        public string VesselId { get; set; } = string.Empty; // could be IMO or external id
        public long AgentId { get; set; }
        public DateTime ArrivalDate { get; set; }
        public DateTime? DepartureDate { get; set; }

        // Navigation collections
        public List<ContainerItem> CargoManifest { get; set; } = new List<ContainerItem>();
        public List<CrewMember> CrewMembers { get; set; } = new List<CrewMember>();

        // Status lifecycle: InProgress, Submitted, PendingApproval, Approved, Rejected
        public string Status { get; set; } = "InProgress";
        public DateTime? SubmissionTimestamp { get; set; }

        // Approval fields
        public long? ApprovedDockId { get; set; }
        public string? RejectionReason { get; set; }
        public DateTime? DecisionTimestamp { get; set; }
        public long? OfficerId { get; set; }
    }
}

File: frontend/src/app/models/vessel-visit-notification.ts
export type VisitStatus = 'InProgress' | 'Submitted' | 'PendingApproval' | 'Approved' | 'Rejected' | 'Cancelled' | 'Pending';

export interface CrewMemberDTO {
  name: string;
  citizenId: string;
  nationality: string;
}

export interface ContainerItemDTO {
  containerCode: string;
  cargoType?: string | null;
  isForUnloading: boolean;
}

// Mirrors backend TodoApi.Models.VesselVisitNotifications.VesselVisitNotificationDTO
export interface VesselVisitNotificationDTO {
  id: number;
  vesselId: string; // IMO or external id
  agentId: number;
  arrivalDate: string; // ISO
  departureDate?: string | null; // ISO
  cargoManifest?: string[] | null; // On read, backend returns string codes
  crewMembers?: CrewMemberDTO[] | null;
  status: string; // backend uses strings e.g. InProgress, Approved
  submissionTimestamp?: string | null; // ISO
  approvedDockId?: number | null;
  rejectionReason?: string | null;
  decisionTimestamp?: string | null; // ISO
  officerId?: number | null;
  submittedByRepresentativeEmail?: string | null;
  submittedByRepresentativeName?: string | null;
}

// Mirrors backend CreateVesselVisitNotificationDTO
export interface CreateVesselVisitNotificationDTO {
  vesselId: string;
  agentId: number; // likely ShippingAgent TaxNumber
  arrivalDate: string; // ISO
  departureDate?: string; // ISO
  cargoManifest?: ContainerItemDTO[];
  crewMembers?: CrewMemberDTO[];
  submittedByRepresentativeEmail?: string;
  submittedByRepresentativeName?: string;
}
